http/typescript/fetch/src/httpCore.ts

```diff
@@ -167,4 +187,53 @@ export class HttpCore implements IHttpCore {
         return url;
     }
 
+    /**
+     * Factory method to create HttpCore instance for url requests not using RequestBuilders
+     * @param coreConfig the configuration used to set authprovider, httpClient or middleware, fetchOptions and more.
+     */
+    public static createCoreForPrimitiveRequests(coreConfig: CoreConfig): HttpCore {
+        const coreOptions: CoreOptions = {
+            httpClient: HttpCore.getHttpClient(coreConfig.httpClient, coreConfig.middleware),
+            authenticationProvider: coreConfig.authenticationProvider,
+            fetchOptions: coreConfig.fetchOptions
+        }
+        return new HttpCore({ parseNodeFactory: ParseNodeFactoryRegistry.defaultInstance, serializationWriterFactory: SerializationWriterFactoryRegistry.defaultInstance }, coreOptions);
+    }
+
+    /**
+     * Factory method to create HttpCore instance to support Kiota Request Builders
+     * @param requestBuilderConfig the configuration used to set ParseNodeFactory and serializationWriterFactory
+     * @param coreConfig the configuration used to set authprovider, httpClient or middleware, fetchOptions and more.
+     */
+    public static createCoreForRequestBuilders(requestBuilderConfig: RequestBuilderConfig, coreConfig: CoreConfig): HttpCore {
+        const coreOptions: CoreOptions = {
+            httpClient: HttpCore.getHttpClient(coreConfig.httpClient, coreConfig.middleware),
+            authenticationProvider: coreConfig.authenticationProvider,
+            fetchOptions: coreConfig.fetchOptions
+        }
+        return new HttpCore(requestBuilderConfig, coreOptions);
+    }
+
+    // TODO : add factory method  return instance with default values.
+
+
+    /**
+     * @private
+     * Static method returning an HttpClient instance which will be used by the HttpCore while making request.
+     * @param httpClient the an HttpClient instance passed as a configuration of the HttpCore
+     * @param middleware a single middleware or an array of middleware used to create an HttpClient
+     */
+    private static getHttpClient(httpClient?: HttpClient, middleware?: Middleware | Middleware[]): HttpClient {
```

Comment:
Shouldn't this method be here https://github.com/microsoft/kiota/blob/main/http/typescript/fetch/src/httpClient.ts instead? Also is there a way we could cleanup the parameters story?

Reply:
I might remove this method when introducing the  HttpClientFactory - https://github.com/microsoft/kiota/blob/enhancement/nikithauc-client-middleware-setup/http/typescript/fetch/src/httpClientFactory.ts. 

The methods and the logic of this class introduced in this PR changes are  temporary. The idea of this PR is to introduce the config object pattern and the factory methods and to get the FetchOptions merged.

Reply:
I'd like things to be aligned as much as possible between languages whenever possible/sensible. I believe this method should belong in the other class as it "creates a client". And functionally speaking be a constructor. Can you look at merging those please?